youtube_dl/extractor/streamcz.py

```diff
@@ -1,105 +1,112 @@
+                fatal=False
+            ))
+
+        # HLS
+        hls_rel_url = get_url('hls')
+        if hls_rel_url:
+            formats.extend(self._extract_m3u8_formats(
+                urljoin(sdn_url, hls_rel_url), video_id, m3u8_id='hls',
+                ext='ts', fatal=False
+            ))
+
+        self._sort_formats(formats)
+        return formats
+
+
+class StreamCZIE(StreamCZBase):
+    _VALID_URL = r'https?://(?:www\.)?stream\.cz/(?:[^/]+)/.+-(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'https://www.stream.cz/kdo-to-mluvi/kdo-to-mluvi-velke-odhaleni-prinasi-novy-porad-uz-od-25-srpna-64087937',
         'info_dict': {
-            'id': '10002447',
+            'id': '64087937',
+            'title': 'Kdo to mluví? Velké odhalení přináší nový pořad už od 25. srpna',
+            'description': 'K ikonickým tvářím obrazovek a filmových pláten neodmyslitelně patří i jejich hlasy. Kdo se ale za nimi skrývá? Sledujte nový seriál ze zákulisí dabingu už od příštího úterý.',
+            'timestamp': 1597730400,
+            'upload_date': '20200818',
+            'duration': 51,
             'ext': 'mp4',
-            'title': 'Kancelář Blaník: Tři roky pro Mazánka',
-            'description': 'md5:3862a00ba7bf0b3e44806b544032c859',
-            'thumbnail': 're:^http://im.stream.cz/episode/537f838c50c11f8d21320000',
-            'duration': 368,
-        },
-    }]
+        }
+    }
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
-        api_path = '/episode/%s' % video_id
 
-        req = sanitized_Request(self._API_URL + api_path)
-        req.add_header('Api-Password', _get_api_key(api_path))
-        data = self._download_json(req, video_id)
+        webpage = self._download_webpage(url, video_id)
+        webpage_data = self._html_search_regex(r'window\.APP_SERVER_STATE\s=\s(\{[^;]*)', webpage, 'webpage_data')
+        webpage_json = self._parse_json(webpage_data, video_id, transform_source=js_to_json)
 
-        formats = []
-        for quality, video in enumerate(data['video_qualities']):
-            for f in video['formats']:
-                typ = f['type'].partition('/')[2]
-                qlabel = video.get('quality_label')
-                formats.append({
-                    'format_note': '%s-%s' % (qlabel, typ) if qlabel else typ,
-                    'format_id': '%s-%s' % (typ, f['quality']),
-                    'url': f['source'],
-                    'height': int_or_none(f['quality'].rstrip('p')),
-                    'quality': quality,
-                })
-        self._sort_formats(formats)
+        video_detail = try_get(webpage_json, lambda x: x['data']['fetchable']['episode']['videoDetail']['data'], dict)
+
+        # metadata
+        title = video_detail.get('name')
+        description = video_detail.get('perex')
+        duration = video_detail.get('duration')
+        published = video_detail.get('publishTime').get('timestamp')
```

Comment:
The `publishTime` isn't known to be a `dict`, so `try_get(video_detail, lambda x: x['publishTime']['timestamp'], int)`